### PR TITLE
feat(dashboard): Add maintenance screen

### PR DESCRIPTION
**Pull request by RubenLWF**
# Description

This PR adds a maintenance screen that will be displayed when SudoSOS is under maintenance.
This information is fetched from the `rootApi.ping()` endpoint.

The maintenance screen will look like this:
image

## Types of changes

- New feature _(non-breaking change which adds functionality)_

### CODE REVIEW

File: apps/dashboard/src/main.ts [hunk 1]
```diff
@@ -84,11 +82,13 @@ app.component('Steps', Steps);
...
+await beforeLoad();
+
+try {
+  app.mount('#app');
+} catch {
+  clearTokenInStorage();
+  const authStore = useAuthStore();
+  authStore.logout();
+  app.mount('#app');
+}
```
Comment (SuperVK) [inline apps/dashboard/src/main.ts]: Okay I think I hindsight I gave you the wrong suggestion in the working moment. I would make this:
```suggestion
beforeLoad().then(() => {
    app.mount('#app');
});
```
or 
```suggestion
await beforeLoad()
app.mount('#app');

```

File: apps/dashboard/src/utils/beforeLoadUtil.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { useSettingsStore } from '@/stores/settings.store';
+import { populateStoresFromToken } from '@sudosos/sudosos-frontend-common';
+import apiService from '@/services/ApiService';
+
+export default async function beforeLoad() {
+
+  const settingsStore = useSettingsStore();
+  await settingsStore.fetchMaintenanceMode();
+
+  await populateStoresFromToken(apiService);
```
Comment (SuperVK) [inline apps/dashboard/src/utils/beforeLoadUtil.ts]: And here catch the possible error from `populateStoresFromToken` and log out if need:
```js
clearTokenInStorage();
const authStore = useAuthStore();
authStore.logout();
```

**Review by SuperVK — CHANGES_REQUESTED** [review]

**Review by SuperVK — APPROVED** [review]

**Comment by JustSamuel** [comment]
Do we also want to include the logic for admin bypass here? Or just keep that at the API for now

File: apps/dashboard/src/main.ts [hunk 3]
```diff
@@ -84,11 +81,6 @@ app.component('Steps', Steps);
...
-    app.mount('#app');
-}).catch(() => {
-    clearTokenInStorage();
-    const authStore = useAuthStore();
-    authStore.logout();
-    app.mount('#app');
-});
+await beforeLoad();
+
+app.mount('#app');
```
Comment (JustSamuel) [inline apps/dashboard/src/main.ts]: Top level await is not allowed, use a `.then` instead.

(this is also why you have a build error)

@SuperVK

**Review by JustSamuel — CHANGES_REQUESTED** [review]

**Review by JustSamuel — APPROVED** [review]

**Comment by RubenLWF** [comment]
Does something similar also need to be made for the POS?

I heard @SuperVK mention something about that a websocket first needs to be implemented in the backend

**Comment by JustSamuel** [comment]
Kinda, the POS is not very often refreshed, so my idea was to have a websocket which indicates if the backend is up.
If you feel like taking a deeper dive in the backend, I have a branch which has an almost working version 👀

**Pull request merged**